ethereum-consensus/src/deneb/blob_sidecar.rs

```diff
@@ -29,21 +29,21 @@ pub struct BlobSidecar<
     pub kzg_commitment_inclusion_proof: Vector<Bytes32, KZG_COMMITMENT_INCLUSION_PROOF_DEPTH>,
 }
 
-// TODO: Convert method to free-standing function
-impl<const BYTES_PER_BLOB: usize, const KZG_COMMITMENT_INCLUSION_PROOF_DEPTH: usize>
-    BlobSidecar<BYTES_PER_BLOB, KZG_COMMITMENT_INCLUSION_PROOF_DEPTH>
-{
-    pub fn verify_blob_sidecar_inclusion_proof(&mut self) -> Result<(), Error> {
-        let blob_index = generalized_index_for_blob_index(self.index)?;
-        let subtree_index = get_subtree_index(blob_index);
+pub fn verify_blob_sidecar_inclusion_proof<const KZG_COMMITMENT_INCLUSION_PROOF_DEPTH: usize>(
+    index: BlobIndex,
+    mut kzg_commitment: KzgCommitment,
```

Comment:
better to take `&mut` here so we don't run around cloning this data all the time